server/grpc/rpc.go

```diff
@@ -273,10 +273,15 @@ func (s *RPC) Done(c context.Context, id string, state rpc.State) error {
 		return err
 	}
 	s.completeChildrenIfParentCompleted(workflow)
-
 	if !model.IsThereRunningStage(currentPipeline.Workflows) {
-		if currentPipeline, err = pipeline.UpdateStatusToDone(s.store, *currentPipeline, model.PipelineStatus(currentPipeline.Workflows), workflow.Stopped); err != nil {
-			logger.Error().Err(err).Msgf("pipeline.UpdateStatusToDone: cannot update workflow final state")
+		if len(state.Error) > 0 {
```

Comment:
Shouldn't the error be set on the workflow level?

Reply:
Maybe it should. But for example, when I have a syntax error in of workflows, the whole pipeline is failed. 

Though when one of workflows in a pipeline is killed with an error, the error is not displayed, I'm not sure why. Example: https://ci.badhouseplants.net/repos/2/pipeline/67

And is this kind of errors already supported for workflows? Cause it didn't seem so to me, but I'm still trying to understand how everything is implemented, so I most probably might be wrong

Reply:
Alright, I'll move it there then

Reply:
Hm, it seems that the error is already there. Just not displayed in UI at all

Reply:
I'm running the latest upstream image here 
```bash
$ curl -Q -i ${WOODPECKER_SERVER}/api/repos/2/pipelines/74 -H "Authorization: Bearer ${WOODPECKER_TOKEN}" | jq '.workflows.[] | select(.pid==2) | .error'
"Service \"wp-01hhrxhkcw3zp5j3m38f8epb8v-0-services-0\" is invalid: [spec.ports[0].name: Required value, spec.ports[1].name: Required value]"
```

Reply:
I've got a feeling that to get it done right, the pipeline page should be redesigned. Because one can't really interact with workflows now. What do you think about that?